folium/map.py

```diff
@@ -403,8 +403,11 @@ def __init__(
         self.options = remove_empty(
             draggable=draggable or None, autoPan=draggable or None, **kwargs
         )
+        # this attribute is not used by Marker, but by GeoJson
+        self.icon: Optional[Icon] = None
```

Comment:
This still fails if people use the `add_child` method to add an icon. I am okay with that, but I think we should be aware of it.

Reply:
Also the `render` method (roughly lines 430-436) was originally rewritten to allow people to use `add_child`. If we go back to icon as a member variable, we should make it consistent throughout the class.